.github/workflows/publish.yml

```diff
@@ -71,7 +77,7 @@ jobs:
     steps:
       - uses: aquasecurity/trivy-action@0.11.2
         with:
-          image-ref: ${{ secrets.RELEASE_DAGGER_ENGINE_IMAGE }}:main
+          image-ref: ${{ secrets.RELEASE_DAGGER_ENGINE_IMAGE }}:${{ github.sha }}
```

Comment:
Note (but you already know this I'm sure): I hit this the other day - this workflow doesn't run in this PR, only on `main` so worth keeping an eye no `main` when this merges to make sure it actually works :scream:

Reply:
This is indeed the scary part of the changes here. I'll see how much I can test before the release tomorrow (in the past I had fully replicated everything in my fork so I could actually e2e test a release). I'll update on how much progress I make there and we can make a call on whether it's worth the risk of merging before release tomorrow or postponing one more (😞, but not actually the end of the world)